test/lib/e2e-utils.ts

```diff
@@ -90,6 +90,22 @@ if (testMode === 'dev') {
   ;(global as any).isNextStart = true
 }
 
+/**
+ * Whether the test is running in dev mode.
+ * Based on `process.env.NEXT_TEST_MODE` and the test directory.
+ */
+export const isNextDev = testMode === 'dev'
+/**
+ * Whether the test is running in deploy mode.
+ * Based on `process.env.NEXT_TEST_MODE`.
+ */
+export const isNextDeploy = testMode === 'deploy'
+/**
+ * Whether the test is running in start mode.
+ * Default mode. `true` when both `isNextDev` and `isNextDeploy` are false.
+ */
+export const isNextStart = !isNextDev && !isNextDeploy
+
```

Comment:
```suggestion
declare global {
  var isNextDev: boolean;
  var isNextDeploy: boolean;
  var isNextStart: boolean;
}
```
You could add something like this and then remove all of the `global as any` and things should be okay. 

I'm not a fan of global things, but that might be a better incremental step

Reply:
I think this is an unnecessary "incremental step". Typing global might give the impression that we should keep supporting this, while ideally we just want to get rid of all global mutations.